My reference topic (#71)
UWC2-APIDOC/to-do-service-public: docs/api/params-for-successful-requests.md

---
layout: page
---

# Parameters for successful requests

This section highlights the required and optional parameters for the operations of the To-Do service.

To get a successful response from the following operations, you must specify at least the `required` properties listed under each task. 

## Add a new task

> | Property| Type|Description|

Reviewer: ```suggestion

**HTTP method**: POST
**URL**: `<server>/task`

| Property| Type|Description|
```
All of these need an example of the URL to call to provide context for the properties in the table. For example, the properties in this table go in the request body, while for the "find" operations, the search parameter goes in the URL.

Maybe add a column for where the property is found, e.g. `query parameter` or `request body` or something like that?

|---|---|---|
|user_id | Number   |Indicates the user-specified numeric value associated with a new task.|
| title | String |Indicates a brief note about the task that you want to add.|
| description | String | Description of the task.   |

> | due_date  | String (format: YYYY-MM-DD)   | Completion date of the task.   |

Reviewer: ```suggestion
| due_date  | String (format: YYYY-MM-DDTHH:MM:SS)   | Completion date of the task.   |
```
Don't forget the time!

> | warning  | String  | Indicates the user-specified note that can impact a task.|

Reviewer: ```suggestion
| warning  | number| When to remind the user of the task, relative to the `due_date`. Negative numbers indicate before the `due_date`, and positive numbers, after. |
```
Check the `task` documentation to confirm this change.

# Parameters for successful requests

> This section highlights the required and optional parameters for the To-Do service.

Reviewer: ```suggestion
This section highlights the required and optional parameters for the operation s of the To-Do service.
```

> To get a successful response, at a minimum, you must specify the properties listed under each task. 

Reviewer: ```suggestion
To get a successful response from the following operations,you must specify at least the `required` properties listed under each task. 
```

> # Add a new task

Reviewer: In a markdown file, there can only be one H1 title, which is the topic title.

Author: Got it! Will update..

## URL: /tasks

| Parameters| Type|Description|
|---|---|---|
|user_id | Integer   | Indicates the user-specified numeric value associated with a new task.|
| title | Alphanumeric | Indicates a brief note about the task that you want to add.|
| description | Varchar | Description of the task.   |
| due_date  | Datetime   | Completion date of the task.   |

> | warning  | Varchar  | Indicates the user-specified note that can impact a task.|

Reviewer: REST APIs typically use [JSON values](https://restfulapi.net/json-data-types/). These look like SQL data types, which, if required, would need to be expressed in a JSON data type, with whatever restrictions necessary to make them compatible. e.g. a DATETIME SQL type, would be stored in JSON as a string, formatted as date: YYYY-MM-... Date values in JSON are often expressed in [ISO 8601 format](https://www.iso.org/iso-8601-date-and-time-format.html).
